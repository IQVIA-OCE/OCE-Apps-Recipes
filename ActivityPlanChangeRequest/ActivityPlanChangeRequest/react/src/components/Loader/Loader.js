import React from 'react';
import { View, StyleSheet } from 'react-native';
import { ApolloProgress } from '@oce-apps/apollo-react-native';
import { useSelector } from 'react-redux';
import { loadingStatusSelector } from '../../store/applicationSlice/applicationSelectors';
import { LOADING_STATUS } from '../../constants/loadingStatus';

const Loader = () => {
  const loadingStatus = useSelector(loadingStatusSelector)

  if (loadingStatus === LOADING_STATUS.PENDING) {
    return (
      <View style={styles.container} testID="loader">
        <ApolloProgress />
      </View>
    )
  } else {
    return null;
  }
};

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    top: 0,
    left: 0,
    width: '100%',
    height: '100%',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'rgba(255, 255, 255, 0.7)'
  }
})
export default Loader;
