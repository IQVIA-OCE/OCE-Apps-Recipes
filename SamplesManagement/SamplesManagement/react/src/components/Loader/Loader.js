import React from 'react';
import { StyleSheet, View } from 'react-native';
import { ActivityIndicator, useTheme } from 'apollo-react-native';

const Loader = () => {
  const theme = useTheme();

  return (
    <View style={[styles.root, { backgroundColor: theme.colors.background }]}>
      <ActivityIndicator animating={true} color={theme.colors.primary} />
    </View>
  );
};

const styles = StyleSheet.create({
  root: {
    flexDirection: 'row',
    width: '100%',
    height: '100%',
    zIndex: 100,
    alignContent: 'center',
    justifyContent: 'center',
    position: 'absolute',
    opacity: 0.7,
  },
});

export default Loader;
