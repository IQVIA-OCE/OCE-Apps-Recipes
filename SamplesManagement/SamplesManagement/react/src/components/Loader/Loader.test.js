import React from 'react';
import Loader from './Loader';
import { render } from '@testing-library/react-native';
import { act } from '@testing-library/react-native';

describe('Loader', () => {
  it('Should render Loader', () => {
    const { getByTestId } = render(<Loader />);

    expect(getByTestId('Loader')).toBeTruthy();
    act(() => jest.runAllTimers());
  });
});
