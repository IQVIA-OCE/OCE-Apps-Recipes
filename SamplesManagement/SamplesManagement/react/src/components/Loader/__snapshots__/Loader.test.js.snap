// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Should render Loader 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "backgroundColor": "rgba(255,255,255,0.7)",
      "flexDirection": "row",
      "height": "100%",
      "justifyContent": "center",
      "position": "absolute",
      "width": "100%",
      "zIndex": 100,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "height": 24,
          "opacity": 1,
          "width": 24,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "height": 24,
              "transform": Array [
                Object {
                  "rotate": "45deg",
                },
              ],
              "width": 24,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "height": 12,
                "overflow": "hidden",
                "width": 24,
              }
            }
          >
            <View
              style={
                Object {
                  "height": 24,
                  "transform": Array [
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "rotate": "-165deg",
                    },
                  ],
                  "width": 24,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "height": 12,
                    "overflow": "hidden",
                    "width": 24,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "borderColor": "#015ff1",
                      "borderRadius": 12,
                      "borderWidth": 2.4,
                      "height": 24,
                      "width": 24,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "height": 24,
              "transform": Array [
                Object {
                  "rotate": "45deg",
                },
              ],
              "width": 24,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "height": 12,
                "overflow": "hidden",
                "top": 12,
                "width": 24,
              }
            }
          >
            <View
              style={
                Object {
                  "height": 24,
                  "transform": Array [
                    Object {
                      "translateY": -12,
                    },
                    Object {
                      "rotate": "345deg",
                    },
                  ],
                  "width": 24,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "height": 12,
                    "overflow": "hidden",
                    "width": 24,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "borderColor": "#015ff1",
                      "borderRadius": 12,
                      "borderWidth": 2.4,
                      "height": 24,
                      "width": 24,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
